modules/navier_stokes/include/functormaterials/LinearFVEnthalpyFunctorMaterial.h

```diff
@@ -0,0 +1,42 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "FunctorMaterial.h"
+
+class SinglePhaseFluidProperties;
+
+/**e
+ * Converts temperature to enthalpy or enthalpy to temperature using
+ * functor material properties. The derivatives are discarded, so can't
+ * be used with AD.
+ */
+class LinearFVEnthalpyFunctorMaterial : public FunctorMaterial
+{
+public:
+  static InputParameters validParams();
+  LinearFVEnthalpyFunctorMaterial(const InputParameters & parameters);
+
+protected:
+  /// Variables, treated as functors, needs to be AD because
+  /// Materials inherit from ADFunctorInterface
+  const Moose::Functor<ADReal> & _pressure;
+  const Moose::Functor<ADReal> & _T_fluid;
+  const Moose::Functor<ADReal> & _h;
```

Comment:
```suggestion
  const Moose::Functor<Real> & _pressure;
  const Moose::Functor<Real> & _T_fluid;
  const Moose::Functor<Real> & _h;
```

Reply:
There is a capability to do this now if I remember correctly (got merged in the break).